### ISSUE TITLE
Custom validation not local to model

### ISSUE THREAD

**Issue by richardwardza** [balderdashy/sails#2863]
Hi, I have 2 models:

model A:
```
module.exports = {
    types: {
        is_date: function(date_string){
            if(date_string === null) {
                return true;
            }
            return moment(date_string, 'DD MMMM YYYY').isValid();
        }
    },
    attributes: {
        id: {
            type: 'integer',
        },
        a_date: {
            type: 'string',
            is_date: true
        }
}
```
and model B:
```
module.exports = {
    types: {
        is_date: function(date_string){
            if(date_string === null) {
                return true;
            }
            return moment(date_string, 'YYYY/MM/DD').isValid();
        }
    },
    attributes: {
        id: {
            type: 'integer',
        },
        a_date: {
            type: 'string',
            is_date: true
        }
}
```
They both need to validate the date in different ways, A needs the date in "05 April 2015" and B as "2105/04/15". 
Depending on how the models are loaded, both of them use either Model A's "is_date" validation or Model B's validation.

The models don not use their own is_date validation but whichever one is loaded first/last.
I know I can workaround this by renaming the validation function (is_date_a and is_date_b) but wouldn't it be better if the validation was 'local' to the model and not global?

**Comment by sailsbot** [comment]
Thanks for posting, @richardwardza.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
